modules/standard/Path.chpl

```diff
@@ -209,13 +212,16 @@ proc commonPath(in paths: string ...?n): string {
   }
 
   if (flag == 1) {
-    prefixArray.remove(pos..prefixArray.size);
+    for i in pos..prefixList.size by -1 do
+      try! prefixList.pop(i);
   } else {
-    prefixArray.remove(minPathLength+1..prefixArray.size);
+    for i in (minPathLength + 1)..prefixList.size by -1 do
+      try! prefixList.pop(i);
     // in case all paths are subsets of the longest path thus pos was never
     // updated
   }
 
+  var prefixArray = prefixList.toArray();
```

Comment:
No need to convert to an array. There is an overload of string.join that takes an iterable. So I expect the following to work:
```chpl
return pathSep.join(prefixList.these()); 
```

Ideally, `these()` would not be required, however we are not there at the moment.